### PR TITLE
Remove src0 src1, rename opt to src

### PR DESCRIPTION
**Pull request by rezaarezvan**
Trying to add #341 - I'll open this draft PR while I try to add/fix everything :)

### CODE REVIEW

**Comment by rezaarezvan** [comment]
It compiles, but running tests makes 6 of the tests segfault. Not good, I'll look into this. 

```
The following tests FAILED:
          1 - test-grad0 (SEGFAULT)
          2 - test-opt (SEGFAULT)
          5 - test-mul-mat0 (SEGFAULT)
          8 - test1 (SEGFAULT)
          9 - test2 (SEGFAULT)
         10 - test3 (SEGFAULT)
```

**Comment by slaren** [comment]
You need to fix the indices of what previously were the `opt` tensors, `opt[0]` should be `src[2]`, `opt[1]` -> `src[3]`, etc.

**Comment by rezaarezvan** [comment]
> You need to fix the indices of what previously were the `opt` tensors, `opt[0]` should be `src[2]`, `opt[1]` -> `src[3]`, etc.

Yes, I'm doing that right now :). I'll send that commit soon!

File: src/ggml.c [hunk 1]
```diff
@@ -5144,9 +5144,9 @@ struct ggml_tensor * ggml_acc_impl(
 
     result->op   = GGML_OP_ACC;
     result->grad = is_node ? ggml_dup_tensor(ctx, result) : NULL;
-    result->src0 = a;
-    result->src1 = b;
-    result->opt[0] = c;
+    result->src[0] = a;
+    result->src[1] = b;
+    result->src[0] = c;
```
Comment (nullhook) [inline src/ggml.c:5149]: why are we re-writing to 0th index here again? perhaps a mistake? This should be src[2]?

Comment (nullhook) [reply src/ggml.c:5149]: additionally, do we need a bounds check here?

Comment (rezaarezvan) [reply src/ggml.c:5149]: > why are we re-writing to 0th index here again? perhaps a mistake? This should be src[2]?

Yes, I'm fixing all indices as soon as possible :).

**Comment by rezaarezvan** [comment]
Still segfaults so, I've not properly managed to fix this - I'll continue :)

**Pull request closed without merging**